include/zephyr/drivers/gpio/rt1718s.h

```diff
@@ -0,0 +1,165 @@
+/*
```

Comment:
I think most of this file (all of it?) should not go in include/zephyr/drivers, but rather in drivers/gpio/gpio_rt1718s.h and be included locally. This should only be used for device specific higher level functions, whatever is not covered by the gpio api.